### DISCUSSION TITLE
@mdx-js/loader not working with Yarn 3

### DISCUSSION THREAD

**Discussion by Pablodotjs** [mdx-js/mdx#1960] [Q&A] [answered]
I have a project with yarn 3, this project has a nextjs app (Next v12), and when trying to integrate MDX to it, and we visit the route where the mdx page is I receive a Module not Found error, it is already installed and everything but it keeps throwing the error. The app runs and you can visit not MDX pages.

Tried with a brand new app outside that repo that does not have yarn 3 and the integration works as expected.

Any help?

**Comment by ChristianMurphy** [comment, marked as answer]
This sounds the same as https://github.com/mdx-js/mdx/issues/1922?
If so, the same fix from https://github.com/yarnpkg/berry/issues/4034 likely applies.

**Reply by ChristianMurphy** [reply]
could you share how you tried to integrate in a sandbox like: https://codesandbox.io/s/github/mdx-js/.github/tree/main/sandbox-templates/mdx-loader-next or https://stackblitz.com/github/mdx-js/.github/tree/main/sandbox-templates/mdx-loader-next ?

**Reply by Pablodotjs** [reply]
Forget my previous answer, I've set nodeLinker: node-modules on .yarnrc.yml to the next js app and after that I runned yarn and now it works, but as there are other apps and stuff that uses the pnp I need to make that rule specifically for that nextjs app because it deleted the .pnp.cjs and .pnp.loader.mjs from the root folder of the repo
